src/js/transfers/bridged-nep21-to-erc20/index.js

```diff
@@ -6,56 +6,78 @@ import { borshifyOutcomeProof } from './borshify-proof'
 import { COMPLETE, FAILED, SUCCESS } from '../statuses'
 import * as storage from '../storage'
 
-// Call contract given by `nep21Address` to withdraw (burn) the given `amount`
-// of NEP21 tokens for window.nearUserAddress. The raw transaction info is
-// returned. This info is then used in `checkStatus` to derive the proof needed
-// for releasing `amount` tokens to window.ethUserAddress on Ethereum.
+// Immediately return to caller, which adds an ID to the provided data, stores
+// it in localStorage, and then calls `checkStatus` (below), which will then
+// call actual `withdraw` function.
 export async function initiate (nep21Address, amount) {
-  const bridgeToken = new NearContract(
-    window.nearConnection.account(),
+  return {
+    amount,
     nep21Address,
-    { changeMethods: ['withdraw'] }
-  )
+    status: INITIALIZED
+  }
+}
 
-  // TODO: can cause redirect to NEAR Wallet, rendering `tx` inaccessible!
-  const tx = await bridgeToken.withdraw(
-    {
-      amount: String(amount),
-      recipient: window.ethUserAddress.replace('0x', '')
-    },
-    new BN('3' + '0'.repeat(14)) // 10x current default from near-api-js
-  )
+async function withdraw ({ id, nep21Address, amount }) {
+  // Calling `bridgeToken.withdraw`, below, may cause a redirect to NEAR
+  // Wallet. Once back at this page, near-api-js will add transaction info to
+  // `completedTransactions` iterator.
+  //
+  // Given that the code here will only be processed when status is
+  // INITIALIZED, if we don't find this tx id in `completedTransactions`, then
+  // we know we need to start the withdrawal.
+  const withdrawing = window.nearConnection
+    .completedTransactions.remove(tx => tx.meta.id === id)
```

Comment:
I think a better API can be all transactions and then providing a status inside:
```suggestion
  const withdrawing = window.nearConnection
    .transactions.remove(tx => tx.meta.id === id && tx.status == TransactionStatus.Completed)
```

Reply:
Do you think the name `cachedTransactions` would be more descriptive? Calling it just `transactions` seems a little cryptic to me.

Reply:
Sticking with `completedTransactions` for now, since that's the only use-case we currently know of, and it avoids needing to import `TransactionStatus.Completed` for the same functionality.